crates/node/src/sync/key.rs

```diff
@@ -53,11 +53,11 @@ impl Node {
 
         let shared_key = SharedKey::new(&private_key, &their_identity);
 
-        let Some(ack) = recv(stream, self.sync_config.timeout, Some(shared_key)).await? else {
-            bail!("connection closed while awaiting state sync handshake");
+        let Some(msg) = recv(stream, self.sync_config.timeout, Some(shared_key)).await? else {
+            bail!("connection closed while awaiting key share");
```

Comment:
I think key share sync should be bi-directional, so before receiving, we should share ours